File: backend/src/main/java/com/coing/domain/chat/repository/ChatMessageRepository.java
package com.coing.domain.chat.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import com.coing.domain.chat.entity.ChatMessage;

public interface ChatMessageRepository extends JpaRepository<ChatMessage, Long> {
	List<ChatMessage> findByChatRoomIdOrderByTimestampAsc(Long chatRoomId);
}

File: backend/src/main/java/com/coing/domain/chat/repository/ChatMessageRepository.kt
package com.coing.domain.chat.repository

import com.coing.domain.chat.entity.ChatMessage
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.time.LocalDateTime

@Repository
interface ChatMessageRepository : JpaRepository<ChatMessage, Long> {
    fun findAllByChatRoomIdAndTimestampAfter(chatRoomId: Long, timestamp: LocalDateTime): List<ChatMessage>
}
